### PR TITLE
Adding deck 1.24 including apiops guide

### PR DESCRIPTION
**Pull request by rspurgeon**
### Description

Adding deck 1.24 release and APIOps guide
 
Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc.

https://konghq.atlassian.net/browse/APIOPS-61

### Testing instructions

Netlify link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

**Comment by rspurgeon** [comment]
Draft PR as of now to allow us to collaborate on the release

File: app/_src/deck/guides/apiops.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+## Configuration Transformations
+
+If you are building microservices or an API platform for multiple teams, you will likely have 
+multiple services and code repositories which may have their own decK configuration files. 
+Using decK file transformation commands, you can organize your decK configuration files into partial segments 
+of the full configuration and assemble them prior to syncronizing with {{site.base_gateway}}. 
+This allows you to organize different aspects of the configuration in alignment with the rest of your
+software development artifacts.
+
+Continuing the example above, let's assume you have second team that builds a different API, and
```
Comment (lena-larionova) [inline app/_src/deck/guides/apiops.md:119]:
```suggestion
Continuing the example above, let's take a look at how commands can be pipelined to create API lifecycle automations.

Let's assume you have second team that builds a different API, and
```
Moving statement from the bottom of the section.

File: app/_src/deck/guides/apiops.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+an Unix style pipeline:
+
+```sh
+deck file openapi2kong --spec oas.yaml --output-file mockbin.yaml && 
+  deck file merge mockbin.yaml another-mockbin.yaml | 
+  deck file patch --selector "$.services[*]" --value 'protocol: "https"' |
+  deck sync -s -
+```
+
+Most commonly you will use the commands from CI/CD tools built into your version control system
```
Comment (lena-larionova) [inline app/_src/deck/guides/apiops.md:184]:
```suggestion
Most commonly, you will use the commands from CI/CD tools built into your version control system
```

**Review by lena-larionova — COMMENTED** [review]
All minor edits, I didn't want to go too deeply into the review in the interest of time.

File: app/_src/deck/guides/apiops.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    plugins: []
+    regex_priority: 200
+    strip_path: false
+    tags: []
+  tags: []
+upstreams: []
+```
+
+{:.note}
+> **Note**: The {{site.base_gateway}} [getting started guide](/gateway/{{page.kong_version}}/get-started/)
```
[the remark this answers is not in the document]
Comment (rspurgeon) [reply app/_src/deck/guides/apiops.md:81]: thanks, just saw that.

**Comment by rspurgeon** [comment]
> All minor edits, I didn't want to go too deeply into the review in the interest of time.

Thank you, all suggestions applied. Given time after release I'm happy to revisit the guide page if we think it needs more work

File: app/_src/deck/guides/apiops.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+```sh
+deck file merge mockbin.yaml another-mockbin.yaml \
+  --output-file merged-kong.yaml
+``` 
+
+You now have a file named `merged-kong.yaml`, which is a single decK file with both services and routes merged. This file is
+also a complete deck file and could be synchronized to a gateway. Before doing that, let's take the example one step further.
+
+Now assume you want to ensure that all services in your configuration communicate with the upstream endpoint 
+via `https` only. We can use the deck `file patch` command to accomplish this.
```
Comment (lena-larionova) [inline app/_src/deck/guides/apiops.md:163]:
```suggestion
via `https` only. You can use the deck `file patch` command to accomplish this:
```

**Review by lena-larionova — COMMENTED** [review]
Couple more minor things I didn't catch, otherwise looks good.

File: app/_src/deck/reference/deck_file_add-plugins.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+:  Specify this flag to overwrite plugins by the same name if they already
+exist in an array. The default behavior is to skip existing plugins. (Default: `false`)
+
+`--selector`
+:  JSON path expression to select plugin-owning objects to add plugins to.
+Defaults to the top level (selector `$`). Repeat for multiple selectors.
+
+{:.important}
+> **Warning**: The JSONPath implementation has a known issue related to 
+recursive descent with expressions. Expressions following a recusrive
```
Comment (lena-larionova) [inline app/_src/deck/reference/deck_file_add-plugins.md:63]:
```suggestion
recursive descent with expressions. Expressions following a recursive
```

**Pull request merged**